File: PaMVPDemo/lib-core/src/main/java/com/pa/core/config/ImageLoaderConfig.java
package com.pa.core.config;

import android.content.Context;

public class ImageLoaderConfig {
    public final String cacheUrl;
    public final int diskCacheSizeBytes;
    public final int memorySize;
    public final Context context;

    private ImageLoaderConfig(Builder builder){
        this.cacheUrl = builder.cacheUrl;
        this.diskCacheSizeBytes = builder.diskCacheSizeBytes;
        this.memorySize = builder.memorySize;
        this.context = builder.context;
    }

    public static class Builder{
        String cacheUrl;
        int diskCacheSizeBytes = 30;
        int memorySize = 30;
        Context context;


        public Builder setCacheUrl(String cacheUrl){
            this.cacheUrl = cacheUrl;
            return this;
        }

        public Builder setDiskCacheSizeBytes(int diskCacheSizeBytes){
            this.diskCacheSizeBytes = diskCacheSizeBytes;
            return this;
        }

        public Builder setMemorySize(int memorySize){
            this.memorySize = memorySize;
            return this;
        }

        public Builder setContexts(Context context){
            this.context = context;
            return this;
        }

        public ImageLoaderConfig build(){
            return new ImageLoaderConfig(this);
        }
    }
}

File: PaMVPKotlinDemo/lib-core/src/main/java/com/pa/core/config/ImageLoaderConfig.kt
package com.pa.core.config

import android.content.Context

class ImageLoaderConfig private constructor(builder: Builder) {
    val cacheUrl: String?
    val diskCacheSizeBytes: Int
    val memorySize: Int
    val context: Context?

    init {
        this.cacheUrl = builder.cacheUrl
        this.diskCacheSizeBytes = builder.diskCacheSizeBytes
        this.memorySize = builder.memorySize
        this.context = builder.context
    }

    class Builder {
        var cacheUrl: String? = null
        var diskCacheSizeBytes = 30
        var memorySize = 30
        var context: Context? = null


        fun setCacheUrl(cacheUrl: String): Builder {
            this.cacheUrl = cacheUrl
            return this
        }

        fun setDiskCacheSizeBytes(diskCacheSizeBytes: Int): Builder {
            this.diskCacheSizeBytes = diskCacheSizeBytes
            return this
        }

        fun setMemorySize(memorySize: Int): Builder {
            this.memorySize = memorySize
            return this
        }

        fun setContexts(context: Context): Builder {
            this.context = context
            return this
        }

        fun build(): ImageLoaderConfig {
            return ImageLoaderConfig(this)
        }
    }
}
